SDL_hidapi_steam: build failure on c2x

Trying to build SDL3 using LLVM 15/16 and `-std=c2x` fails with the following error:

```
[141/381] Building C object CMakeFiles/SDL3-shared.dir/src/joystick/hidapi/SDL_hidapi_steam.c.o
FAILED: CMakeFiles/SDL3-shared.dir/src/joystick/hidapi/SDL_hidapi_steam.c.o 
/opt/homebrew/opt/llvm/bin/clang -DSDL3_shared_EXPORTS -DSDL_BUILD_MAJOR_VERSION=3 -DSDL_BUILD_MICRO_VERSION=0 -DSDL_BUILD_MINOR_VERSION=0 -DUSING_GENERATED_CONFIG_H -I/SDL/build/include-config-release -I/SDL/src -I/SDL/build/include -I/SDL/include -I/SDL/include/SDL3 -std=c2x -O3 -DNDEBUG -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk -fPIC -fvisibility=hidden -Wno-error=deprecated-declarations -D_THREAD_SAFE -fobjc-arc -idirafter /SDL/src/video/khronos -Wall -fno-strict-aliasing -Wdeclaration-after-statement -Wdocumentation -Wdocumentation-unknown-command -fcomment-block-commands=threadsafety -Wshadow -fcolor-diagnostics -Winvalid-pch -Xarch_arm64 -include/SDL/build/CMakeFiles/SDL3-shared.dir/cmake_pch_arm64.h -MD -MT CMakeFiles/SDL3-shared.dir/src/joystick/hidapi/SDL_hidapi_steam.c.o -MF CMakeFiles/SDL3-shared.dir/src/joystick/hidapi/SDL_hidapi_steam.c.o.d -o CMakeFiles/SDL3-shared.dir/src/joystick/hidapi/SDL_hidapi_steam.c.o -c /SDL/src/joystick/hidapi/SDL_hidapi_steam.c
/SDL/src/joystick/hidapi/SDL_hidapi_steam.c:36:5: error: expected identifier
    false,
    ^
/SDL/src/joystick/hidapi/SDL_hidapi_steam.c:37:5: error: expected identifier
    true
    ^
/SDL/src/joystick/hidapi/SDL_hidapi_steam.c:38:2: error: expected ';' after enum
} bool;
 ^
 ;
/SDL/src/joystick/hidapi/SDL_hidapi_steam.c:34:1: warning: typedef requires a name [-Wmissing-declarations]
typedef enum
^~~~~~~
/SDL/src/joystick/hidapi/SDL_hidapi_steam.c:38:3: warning: declaration does not declare anything [-Wmissing-declarations]
} bool;
  ^~~~
```

It seems to be related to the enum in `src/joystick/hidapi/SDL_hidapi_steam.c`:

```c
typedef enum {
  false,
  true
} bool;
```

C23 make `true` and `false` keywords, which is implemented in LLVM Clang since version 15.
Unfortunately, I do not see any feature check that we could use for falling back to the enum. I will see about using `check_c_source_compiles` to test for that.

---

This affects:
- main branch @ 586004e52a7bb81ab970e58804620dec4e3b1d70
- SDL2 branch @ b57a6a1adf15f9a6e4a48e1980aa723d47fa2fe3
- release-2.26.x branch @ 24098b75b10ec827d517715d0fb963b0bcce7c3c

--- a/src/joystick/hidapi/SDL_hidapi_steam.c
+++ b/src/joystick/hidapi/SDL_hidapi_steam.c
@@ -31,11 +31,9 @@
 
 #include <stdint.h>
 
-typedef enum
-{
-    false,
-    true
-} bool;
+#define bool SDL_bool
+#define true SDL_TRUE
+#define false SDL_FALSE
 
 typedef uint32_t uint32;
 typedef uint64_t uint64;

